lib/logstash/filters/elasticsearch.rb

```diff
@@ -121,7 +121,7 @@ def filter(event)
       @logger.warn("Failed to query elasticsearch for previous event", :index => @index, :query => query, :event => event, :error => e)
       @tag_on_failure.each{|tag| event.tag(tag)}
     end
-    filter_matched(event)
+    filter_matched(event) if !resultsHits.nil? && !resultsHits.empty?
```

Comment:
not sure if this is the ideal criteria. First, we also have [the result of aggregations](https://github.com/bperian/logstash-filter-elasticsearch/blob/cc6455d28217b560db7ee96f0313766c49a4334e/lib/logstash/filters/elasticsearch.rb#L113) which may not be empty. If an aggregation is requested with `size: 0` then no hits will be returned, only aggregations. Maybe we should do something like:

```ruby
# check if the hits array exists and if there are hits
# if the query is an aggregation using the query dsl with size 0 we'll still see total hits > 0
filter_matched(event) if (resultsHits && results["hits"]["total"] > 0)
```

Reply:
Thank you for the feedback. The lack of that condition affects any action that is based on the result of filter matched ( tagging, fields, etc. ) from my tests. I'll get back with changes asap that will acommodate all feedback ( and also do thorough testing)